### DISCUSSION TITLE
Auth abilities / Webhook Build hooks

### DISCUSSION THREAD

**Discussion by williamjulianvicary** [coollabsio/coolify#2971] [Feature Requests]
Build webhooks require auth, which is OK however a few issues with that:

- It requires a write token - which requires the API to be enabled and gives access to all resources in essentially root access.

Proposed adjustment:

- Allow optional non-authenticated webhooks (URL based tokens for URL based deploys)
- Allow abilities to be constrained so (at least) deploy hooks can be constrained to "Deploy only" rather than requiring a write token that has root capabilities over all Coolify features.

**Comment by andrasbacsai** [comment]
We will add `deploy` and `root` token permission in v4.0.0-beta.377.

So you could have `root`, `write`, `read`, `read:sensitive` and `deploy` permissions.

- root - Access to everything.
- write - Access to endpoints which are modifying someting in the database.
- read - Read endpoints without sensitive data.
- read:sensitive - Read endpoints with sensitive data.
- deploy - Only able to call `/deploy` endpoint.
